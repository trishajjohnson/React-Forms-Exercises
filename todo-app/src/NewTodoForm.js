import React, { useState } from "react";
import { v4 as uuid } from "uuid";

const NewTodoForm = ({createTodo}) => {
    const [task, setTask] = useState("");

    const handleChange = (e) => {
        setTask(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        createTodo( {task, id: uuid() });
        setTask("");
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="task">Task</label>
            <input 
              id="task" 
              type="text" 
              name="task"
              value={task} 
              onChange={handleChange} 
            />

            <button>Add Todo</button>
        </form>
    );
}

export default NewTodoForm;