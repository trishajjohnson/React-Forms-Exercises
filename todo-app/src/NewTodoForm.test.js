import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

// Smoke test for NewTodoForm component
it("should render without crashing", function() {
  render(<NewTodoForm />)
});

// Snapshot test for NewTodoForm component
it("should match snapshot", function() {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it("runs the create function on form submit", function() {
    const createMock = jest.fn();
    const { getByText } = render(<NewTodoForm createTodo={createMock} />);
    const addBtn = getByText("Add Todo");
    fireEvent.click(addBtn);
    expect(createMock).toHaveBeenCalled();
});