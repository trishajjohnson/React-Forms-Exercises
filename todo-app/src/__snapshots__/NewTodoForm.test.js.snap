// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Task
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
